protocol/0014-ORDT-order_types.md

```diff
@@ -49,6 +49,117 @@ Notes on scope of current version of this spec:
 1. **Reduce-Only (True/False):** Only valid for Non-Persistent orders. Cannot be True at the same time as Post-Only. If set, order will only be executed if the outcome of the trade moves the trader's position closer to 0. In addition, a Reduce-Only order will not move a position to the opposite side to the trader's current position (e.g. if short, a Reduce-Only order cannot make the trader long as a result). If submitted as IOC, where the full volume would switch sides, only the amount required to move the position to 0 will be executed.
 
+
+### 'Glassberg' transparent iceberg orders
+
+On centralised exchanges, icebergs are a type of order that enables a trader to make an order with a relatively small visible "display quantiy" and a larger hidden total size.
+Like an iceberg, most of the order is not visible to other traders.
+After the full size of the visible portion of the order has traded away, the order is "refreshed" and reappears with its maximum display quantity.
+This is repeated until the order is cancelled or expires, or its full volume trades away.
+Some platforms also allow the display quantity to be randomised on each refresh to further frustrate traders trying to identify the existence of the iceberg ordeer).
+
+Vega, being a decentralised and fully transparent protocol, cannot (in its current form) achieve the hidden characteristic of iceberg orders.
+But it can do the rest hence, _glassberg_ orders.
+These are still helpful, especially for market makers and in combination with pegged orders, as they allow a trader to remain competitively present on the order book through the refresh facility without needing to supply excessive volume into a large aggressive order.
+
+
+#### Definitions
+
+These terms are used to refer to fields on an order:
+
+* `quantity` - the full initial size of the order on entry.
+
+* `displayed quantity` - the current displayed quantity, i.e. the amount of the remaining quantity that is active on the book and can be hit.
+Note that for a non-glassberg order, `displayed quantity == remaining`.
+
+* `remaining` - the total quantity of the order remaining that could trade.
+
+
+#### Creating glassberg orders
+
+Glassberg orders are created by populating three additional fields on any valid persistent limit order:
+
+* `initial peak size` - this specifies the amount displayed and available on the order book for a new or newly refreshed glassberg order.
+
+* `minimum peak size` - this determines when a glassberg order is eligible for refresh.
+The glassberg is refreshed any time the order's displayed quantity less than the minimum peak size.
+
+* `refresh policy` - this specifies when a glassberg order that is eligible for refresh and has remaining volume >0 is refreshed:
+
+    * `IMMEDIATE` - the refresh occurs _after_ processing the transaction that depleted the display quantity to zero.
```

Comment:
Thought, how does this interact with batch transactions? Our current rule is that multiple orders in a batch should execute as if each was an individual transaction. However with glassbergs this would mean that, for immediate refresh ones, you might well get a better price by splitting your order up into as many pieces as the batch allows. Do we care or just assume that’s still not a huge number of orders so probably ok?

Reply:
I think that's ok; as the submitter of the glassberg has the option to set `BLOCK_END` if they don't want this.

Reply:
I think we have two issues I might like to prevent that this has highlighted to me:

1. If an `IMMEDTIATELY` refreshing glassberg (assuming not post only) is crossed and trades on entry it will refresh and trade again. For a large quantity/small peak against significant volume this could create a large number of loops. Better to treat a glassberg as its full remaining size on entry during aggressive execution (this does not conflict with "disaplayed" quantity as aggressive size that trades on entry is never displayed on the book).

2. Although this is different to the usual policy for batch orders, perhaps we should make clear that glassberg refreshes happen at transaction end (_strictly_ as specified). I think it is a fairness violation / abuse of batch orders to be able to atomically take more than the `initial peak size` in a single transaction.

Thoughts?

Reply:
I updated to include the above btw, please shout if in disagreement with this